src/validate.js

```diff
@@ -159,6 +160,449 @@ function transformAjvErrors(errors = []) {
   });
 }
 
+/**
+ * Recursively searches through the _errorSchema to find all `__errors` entries.
+ * When one is found a new property is added to the `_target` with the path to
+ * this `__errors` as the key and the value of `__errors`.
+ *
+ * @example
+ * // sample input
+ * {
+ *     foo: {
+ *         bar: {
+ *             __errors: [...__errorsOfThisProp],
+ *             2: {
+ *                 __errors: [...__errorsOfThisProp]
+ *             }
+ *         }
+ *     }
+ * }
+ * // sample output
+ * // different cases possible of an error.properties; here are some EXAMPLES
+ * {
+ *     ".foo": [* errors like: a -> is-required; is-not-a-string; is-not-of-type *],
+ *     ".bar[n]": [* errors like: b -> is-not-an-array *],
+ *     ".bar[n].qux": [* errors like: (a) *],
+ *     ".bar[n].qux[4]": [* errors like: (b) *],
+ *     ".bar[n].qux[4].wibble": [* errors like: (a) *],
+ *     "": [* errors like: does-not-match-oneof-schema *]
+ * }
+ *
+ * @param {object} _target - the object to add the key to.
+ * @param {object[]} _errors - the errors of ajv
+ * @param {object} _errorSchema - the error schema constructed by the `toErrorSchema` function
+ * @param {string} _currentPath - the path of the current nested object
+ */
+function flattenErrorSchema(_target, _errors, _errorSchema, _currentPath = "") {
```

Comment:
It's kind of complicated, it's one of the first things I added :disappointed:. 

But I didn't want to introduce new library dependencies like https://www.npmjs.com/package/flat to add to the bundle size, but If you say I can add it, I'd check it out if it's usable.

Reply:
Yeah, if it's possible to use a library to do this, I'd say simplifying the complexity is worth the increase in bundle size.

Reply:
I tried using this lib but I cannot easily use it because it does not care about arrays and this function also flattens the error schema path depending on the given errors.

I would like to be able to distinguish between `foo[0].bar` and `foo.0.bar`.

Reply:
Why do you need to be able to distinguish between foo[0].bar and foo.0.bar?

Reply:
Good question. I believe I did it like this because it's easier to figure out that `bar` is a property to look for in the uiSchema, and not `0`. 

ATM when it's foo[0].bar I know that foo is an array and "bar" would appear as it's own property in the matching uiSchema path, so I can simply access that property. When it's foo.0.bar I know foo is an object, with a property that is actually named `0` so I need to look that up in the uiSchema.

If it always would be flattened like foo.0.bar I would not have this info. Then I need to look up `0` and if it is not given, `bar` in the uiSchema.

Does this make any sense?

Reply:
I realized I can achieve the same thing by iterating over all errors and combining all errors for one faulty schema property (`error.property`).

Should be much easier now!